In [5]:
!pip install pandas

In [6]:
import pandas as pd

In [8]:
intakes = pd.read_csv('data/aac_intakes.csv')

In [12]:
intakes.head(1)

age_upon_intake animal_id animal_type                     breed  \
0         8 years   A706918         Dog  English Springer Spaniel   

         color                 datetime                datetime2  \
0  White/Liver  2015-07-05T12:59:00.000  2015-07-05T12:59:00.000   

                     found_location intake_condition intake_type   name  \
0  9409 Bluegrass Dr in Austin (TX)           Normal       Stray  Belle   

  sex_upon_intake  
0   Spayed Female

In [9]:
outcomes = pd.read_csv('data/aac_outcomes.csv')

In [15]:
outcomes.head(1)

age_upon_outcome animal_id animal_type                   breed  \
0          2 weeks   A684346         Cat  Domestic Shorthair Mix   

          color        date_of_birth             datetime  \
0  Orange Tabby  2014-07-07T00:00:00  2014-07-22T16:04:00   

             monthyear name outcome_subtype outcome_type sex_upon_outcome  
0  2014-07-22T16:04:00  NaN         Partner     Transfer      Intact Male

In [17]:
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_intake   80187 non-null  object
 1   animal_id         80187 non-null  object
 2   animal_type       80187 non-null  object
 3   breed             80187 non-null  object
 4   color             80187 non-null  object
 5   datetime          80187 non-null  object
 6   datetime2         80187 non-null  object
 7   found_location    80187 non-null  object
 8   intake_condition  80187 non-null  object
 9   intake_type       80187 non-null  object
 10  name              55603 non-null  object
 11  sex_upon_intake   80186 non-null  object
dtypes: object(12)
memory usage: 7.3+ MB


In [18]:
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80681 entries, 0 to 80680
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_outcome  80673 non-null  object
 1   animal_id         80681 non-null  object
 2   animal_type       80681 non-null  object
 3   breed             80681 non-null  object
 4   color             80681 non-null  object
 5   date_of_birth     80681 non-null  object
 6   datetime          80681 non-null  object
 7   monthyear         80681 non-null  object
 8   name              56116 non-null  object
 9   outcome_subtype   36893 non-null  object
 10  outcome_type      80667 non-null  object
 11  sex_upon_outcome  80679 non-null  object
dtypes: object(12)
memory usage: 7.4+ MB


**1) Top Five Places Where Animals Are Found**

In [14]:
intakes.groupby('found_location').size().sort_values(ascending=False)[:5]

found_location
Austin (TX)                          14443
Outside Jurisdiction                   948
Travis (TX)                            921
7201 Levander Loop in Austin (TX)      517
Del Valle (TX)                         411
dtype: int64

I got this answer by grouping the values by the 'found_location' variable, counting the number of rows for each grouping, sorting the counts in descending order and then finding the first five results.

The answer is Austin (TX), Outside Jurisdiction, Travis(TX), 7201 Levander Loop in Austin (TX), and Del Valle (TX)

**2) What is the average number of pets found in a month in the year 2015? Are there months where there is a higher number of animals found?**

In [17]:
intakes['year_found'] = pd.DatetimeIndex(intakes['datetime']).year

In [18]:
intakes['month_found'] = pd.DatetimeIndex(intakes['datetime']).month

In [41]:
intakes_by_month = intakes.loc[intakes['year_found']==2015].groupby('month_found').size()

In [37]:
intakes_by_month.mean()

1559.3333333333333

In [43]:
intakes_by_month > 1560

month_found
1     False
2     False
3     False
4     False
5      True
6      True
7      True
8      True
9      True
10     True
11    False
12    False
dtype: bool

**The mean number of intakes per month in 2015 is 1559.3333333333333 or 1560 pets. The months where the number of intakes were greater than this average were May to October 2015. I found these answers by creating a new data frame called intakes_by_month that had the number of intakes per month in 2015. I then found the average of these counts using the mean() function and then used a comparison operator to figure out which months had counts that were greater than 1560.** 

**3) Ratio of incoming pets versus adopted pets** going forward: find out the different values in outcome to see if there is a specific adopted category and then divide all the intakes by the number of pets adopted if that number exists. 

In [45]:
intakes.count()

age_upon_intake     80187
animal_id           80187
animal_type         80187
breed               80187
color               80187
datetime            80187
datetime2           80187
found_location      80187
intake_condition    80187
intake_type         80187
name                55603
sex_upon_intake     80186
year                80187
month_found         80187
year_found          80187
dtype: int64

In [46]:
outcomes.count()

age_upon_outcome    80673
animal_id           80681
animal_type         80681
breed               80681
color               80681
date_of_birth       80681
datetime            80681
monthyear           80681
name                56116
outcome_subtype     36893
outcome_type        80667
sex_upon_outcome    80679
dtype: int64

In [49]:
# sorting by first name
intakes_sorted = intakes.sort_values("animal_id", inplace = True)
 


In [51]:
# dropping ALL duplicate values
intakes.drop_duplicates(subset ="animal_id",
                     keep = False, inplace = True)

In [52]:
intakes.count()

age_upon_intake     66211
animal_id           66211
animal_type         66211
breed               66211
color               66211
datetime            66211
datetime2           66211
found_location      66211
intake_condition    66211
intake_type         66211
name                42119
sex_upon_intake     66210
year                66211
month_found         66211
year_found          66211
dtype: int64